src/web/client/extension.ts

```diff
@@ -268,12 +269,95 @@ export function processWillSaveDocument(context: vscode.ExtensionContext) {
     );
 }
 
-export function processWillStartCollaboartion() {
+export function processWillStartCollaboartion(context: vscode.ExtensionContext) {
     if (isCoPresenceEnabled()) {
-        // TODO: Add copresence logic
+        processOnTabChange(context);
+        createWebWorkerInstance(context);
     }
 }
 
+let worker: Worker;
+
+export function processOnTabChange(context: vscode.ExtensionContext) {
+    try {
+        context.subscriptions.push(
+            vscode.window.tabGroups.onDidChangeTabs((event) => {
+                event.opened.concat(event.changed).forEach(tab => {
+                    if (tab.input instanceof vscode.TabInputCustom || tab.input instanceof vscode.TabInputText) {
+                        const document = tab.input;
+                        const entityId = getFileEntityId(document.uri.fsPath);
+                        if (entityId) {
+                            worker.postMessage({
+                                afrConfig: {
+                                    swpId: WebExtensionContext.sharedWorkSpaceMap.get(
+                                        Constants.sharedWorkspaceParameters.SHAREWORKSPACE_ID
+                                    ) as string,
+                                    swptenantId: WebExtensionContext.sharedWorkSpaceMap.get(
+                                        Constants.sharedWorkspaceParameters.TENANT_ID
+                                    ) as string,
+                                    discoveryendpoint: WebExtensionContext.sharedWorkSpaceMap.get(
+                                        Constants.sharedWorkspaceParameters.DISCOVERY_ENDPOINT
+                                    ) as string,
+                                    swpAccessToken: WebExtensionContext.sharedWorkSpaceMap.get(
+                                        Constants.sharedWorkspaceParameters.ACCESS_TOKEN
+                                    ) as string,
+                                },
+                                entityId,
+                            });
+                        }
+                    }
+                });
+            })
+        );
+    } catch (error) {
+        // TODO: add Telemetry
+    }
+}
+
+export function createWebWorkerInstance(
+    context: vscode.ExtensionContext
+) {
+    const webworkerMain = vscode.Uri.joinPath(
+        context.extensionUri,
+        "dist/web/webworker.worker.js"
+    );
```

Comment:
nit: Use this instead  vscode.Uri.joinPath(context.extensionUri, "dist", "web", "webworker.worker.js"); to amke sure we are compatible on all OS systems.

Reply:
Done: https://github.com/microsoft/powerplatform-vscode/pull/745/commits/64f0fa3923d2cf21021c864bdd64e909ee953ac6